.github/scripts/label_promoted_commits.py

```diff
@@ -57,8 +70,25 @@ def main():
             }
             response = requests.get(search_url, headers=headers, params=params)
             prs = response.json().get("items", [])
-            for pr in prs:
-                refs_match = re.findall(r'Refs (?:#|https.*?)(\d+)', pr["body"])
+            if prs:
```

Comment:
i am not against duplicating. but i am lost when reading the duplicated part. could you please add some comment on what use case this branch is supposed to handle?